metricbeat/module/vsphere/host/host.go

```diff
@@ -240,3 +221,65 @@ func getAssetNames(ctx context.Context, pc *property.Collector, hs *mo.HostSyste
 		outputVmNames:      outputVmNames,
 	}, nil
 }
+
+func (m *HostMetricSet) getPerfMetrics(ctx context.Context, perfManager *performance.Manager, hst mo.HostSystem, metricIds []types.PerfMetricId) (map[string]interface{}, error) {
+	metricMap := make(map[string]interface{})
+	summary, err := perfManager.ProviderSummary(ctx, hst.Reference())
+	if err != nil {
+		return metricMap, fmt.Errorf("failed to get summary: %w", err)
+	}
+
+	period := m.Module().Config().Period
+	refreshRate := int32(period.Seconds())
+	if period.Seconds() == LiveInterval {
+		if summary.CurrentSupported {
+			refreshRate = summary.RefreshRate
+			if int32(period.Seconds()) != refreshRate {
+				m.Logger().Warnf("User-provided period %v does not match system's refresh rate %v. Risk of data duplication. Consider adjusting period.", period, refreshRate)
```

Comment:
In this case we are warning the user, but anyhow internally we are using the system set refresh rate. 
So shouldn't we inform the user that we are not using your period, but the internal refreshrate.
With the current warning, the user might think that the period he configured is being used.

Reply:
We will consistently display this warning message:
```
User-provided period %v does not match system's refresh rate %v. Risk of data duplication. Consider adjusting period.
```

Since the fetch period itself remains constant, informing the user about the refresh interval may not be beneficial in this scenario. Additionally, we are already warning the user about the risk of data duplication.

Reply:
Eg: 
User input period: 20s
Refresh rate: 300s
Intervalid gets set to: 300s
Duplicated data gets set till the refreshrate is reached. Is that correct ?

Reply:
Ok, user can then makeout from the warning message. Lets leave it as it is